### ISSUE TITLE
[CI] Reassess value of load tests

### ISSUE THREAD

**Issue by djaglowski** [open-telemetry/opentelemetry-collector-contrib#9107]
> @open-telemetry/collector-contrib-approvers, @open-telemetry/collector-approvers, the load tests have always been a source of test failures for us. All we always do is increase the limits once we start hitting them more frequently. I don't remember ever seeing a true positive out of those tests. Are those tests providing value to us? Or are they just sources of noise?
>
> _Originally posted by @jpkrohling in https://github.com/open-telemetry/opentelemetry-collector-contrib/issues/9094#issuecomment-1090300617_

**Comment by codeboten** [comment]
We've talked about this in the past and there was some discussion of running the load tests less frequently, maybe on a daily basis. I'd prefer continuing to run the load tests on every build, record their results, and if possible, emit some sort of report (ie. codecov) instead of having intermittent failures in the build.

I think it's more important to be able to see the historical results of the load tests, since the intermittent failures are being worked around by bumping the resource requirements every few weeks and hide actual build failures.

I'd be happy to take on a task to produce a report with results from every build into main if folks think it's valuable. Producing the report that's published using github pages has been done in the [python repo](https://open-telemetry.github.io/opentelemetry-python/benchmarks/) for some time, i don't know if there's an easy way to produce a report that can be added as a comment, i'd have to dig into that.

**Comment by djaglowski** [comment]
I fully agree with @codeboten's assessment and suggestion. It's valuable information but can be non-blocking without much risk.

**Comment by jpkrohling** [comment]
If we end up having something like the Python repo, the value from the load tests becomes huge.

**Comment by TylerHelmuth** [comment]
As part of an [ongoing discussion in the Go SIG](https://github.com/open-telemetry/opentelemetry-go/issues/2791) around performance tests, I have requested a machine so that we can test out if private hardware will solve test inconsistencies.  I plan to experiment with the Contrib's load test as well.  

If private hardware is a solution, then hopefully we can find a way to run the tests on the machines going forward.  More discussion on long term plans can be found in the linked issue.

**Comment by dmitryax** [comment]
We ran into an issue when the load tests haven't detected a significant spike. I submitted another issue for the same problem with some suggested action items https://github.com/open-telemetry/opentelemetry-collector-contrib/issues/20718 PTAL
